app/models/material.rb

```diff
@@ -8,7 +8,7 @@ class Material < ApplicationRecord
 
   validates   :name, presence: true
 
-  after_initialize :generate_uuid
+  after_initialize :generate_uuid, if: :new_record?
 
```

Comment:
You could want in future to create Materials with a specific uuid, but that specific uuid will be overwritten by a different uuid with generate_uuid. Would it be better to generate the uuid if it is empty in the new record instead of whenever you create a new record?